OMEdit/OMEdit/OMEditGUI/Annotations/TextAnnotation.cpp

```diff
@@ -300,96 +300,88 @@ void TextAnnotation::drawTextAnnotaion(QPainter *painter)
   applyLinePattern(painter);
   /* Don't apply the fill patterns on Text shapes. */
   /*applyFillPattern(painter);*/
-  qreal dx = ((-boundingRect().left()) - boundingRect().right());
-  qreal dy = ((-boundingRect().top()) - boundingRect().bottom());
-  // first we invert the painter since we have our coordinate system inverted.
-  painter->scale(1.0, -1.0);
-  painter->translate(0, dy);
+  // store the existing transformations
+  const QTransform painterTransform = painter->transform();
+  const qreal scaleX = painterTransform.m11();
+  const qreal scaleY = painterTransform.m22();
+  const qreal shearX = painterTransform.m21();
+  const qreal shearY = painterTransform.m12();
+  // set new transformation for the text based on rotation or scale.
+  qreal sx, sy;
+  if (painterTransform.type() == QTransform::TxRotate) {
```

Comment:
I think we should have the calculation of sx and sy as a separate method: setTransformCalculateScaleXY

Reply:
Where we pass sx and sy as parameters

Reply:
Or have one method that takes one parameter and returns the value for the other

Reply:
Then we could make two calls to a generic function since they kinda depend on each other

Reply:
I m not in favor of adding methods/functions that are not reusable. There is no point of moving this functionality to a new method/function. If the existing function becomes too large or complicated then maybe we can split it but here its just no reason.